### PR TITLE
Update license header and extend pre-commit

### PR DESCRIPTION
**Pull request by kse3hi**
ADO#17300

### CODE REVIEW

**Review by mml5bg — DISMISSED** [review]
LGTM

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,15 +1,3 @@
...
-// *
-// * See the NOTICE file(s) distributed with this work for additional
-// * information regarding copyright ownership.
-// *
-// * This program and the accompanying materials are made available under the
-// * terms of the Apache License 2.0 which is available at
-// * http://www.apache.org/licenses/LICENSE-2.0
-// *
-// * SPDX-License-Identifier: Apache-2.0
-// ********************************************************************************/
```
Comment (MP91) [inline .devcontainer/devcontainer.json:12]: why is this removed? can't we license this file?

File: setup.cfg [hunk 2]
```diff
@@ -13,6 +13,9 @@ check_untyped_defs = True
 install_types = True
 non_interactive = False
 namespace_packages = True
+ignore_missing_imports = True
+implicit_optional = True
+ignore_errors = True
```
Comment (MP91) [inline setup.cfg:18]: is the ignoring of errors and missing imports a good idea? isn't that exactly what we want to find?

File: src/licensevalidator/lib/workflowlicenses.py [hunk 3]
```diff
@@ -159,7 +162,7 @@ def __get_license_for_action(action_repo: str, github_token: str) -> Optional[st
     Returns:
         str: The name of the license, if available.
     """
-    request_headers={"Accept": "application/vnd.github.v3+json"}
+    request_headers = {"Accept": "application/vnd.github.v3+json"}
```
Comment (MP91) [inline src/licensevalidator/lib/workflowlicenses.py:165]: in all other project we force the space between the assignments, please adjust the settings accordingly

**Review by MP91 — CHANGES_REQUESTED** [review]

**Review by MP91 — APPROVED** [review]
lgtm

**Pull request merged**